drivers/storage/portworx/component/prometheus.go

```diff
@@ -0,0 +1,514 @@
+					Verbs:     []string{"get", "list", "watch"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"configmaps"},
+					Verbs:     []string{"get"},
+				},
+				{
+					NonResourceURLs: []string{"/metrics", "/federate"},
+					Verbs:           []string{"get"},
+				},
+			},
+		},
+		ownerRef,
+	)
+}
+
+func (c *prometheus) createOperatorClusterRoleBinding(
+	clusterNamespace string,
+	ownerRef *metav1.OwnerReference,
+) error {
+	return k8sutil.CreateOrUpdateClusterRoleBinding(
+		c.k8sClient,
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            PrometheusOperatorClusterRoleBindingName,
+				OwnerReferences: []metav1.OwnerReference{*ownerRef},
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      PrometheusOperatorServiceAccountName,
+					Namespace: clusterNamespace,
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind:     "ClusterRole",
+				Name:     PrometheusOperatorClusterRoleName,
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+		},
+		ownerRef,
+	)
+}
+
+func (c *prometheus) createPrometheusClusterRoleBinding(
+	clusterNamespace string,
+	ownerRef *metav1.OwnerReference,
+) error {
+	return k8sutil.CreateOrUpdateClusterRoleBinding(
+		c.k8sClient,
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            PrometheusClusterRoleBindingName,
+				OwnerReferences: []metav1.OwnerReference{*ownerRef},
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      PrometheusServiceAccountName,
+					Namespace: clusterNamespace,
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind:     "ClusterRole",
+				Name:     PrometheusClusterRoleName,
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+		},
+		ownerRef,
+	)
+}
+
+func (c *prometheus) createOperatorDeployment(
+	cluster *corev1alpha1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	existingDeployment := &appsv1.Deployment{}
+	err := c.k8sClient.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      PrometheusOperatorDeploymentName,
+			Namespace: cluster.Namespace,
+		},
+		existingDeployment,
+	)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	if !c.isOperatorCreated || errors.IsNotFound(err) {
+		deployment := getPrometheusOperatorDeploymentSpec(cluster, ownerRef)
+		if err = k8sutil.CreateOrUpdateDeployment(c.k8sClient, deployment, ownerRef); err != nil {
+			return err
+		}
+	}
+	c.isOperatorCreated = true
+	return nil
+}
+
+func getPrometheusOperatorDeploymentSpec(
+	cluster *corev1alpha1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) *appsv1.Deployment {
+	replicas := int32(1)
+	runAsNonRoot := true
+	runAsUser := int64(65534)
+	labels := map[string]string{
+		"k8s-app": PrometheusOperatorDeploymentName,
+	}
+	operatorImage := util.GetImageURN(cluster.Spec.CustomImageRegistry, "quay.io/coreos/prometheus-operator:v0.29.0")
```

Comment:
shouldn't the image name be a part of the spec ?

Reply:
as prometheus is external to PX, I think we will eventually have to give more control to the users as to how they want to configure prometheus. May be just take the operator deployment spec in the spec.Monitoring.Prometheus.Spec value.

Reply:
For now, let's not expose it to the users as we also create Custom resources that are registered by the prometheus operator. Allowing change in version could result in incompatible CRD specs. If there is a need we can expose it through env var in the near future.